xarray/core/computation.py

```diff
@@ -1384,8 +1383,10 @@ def _calc_idxminmax(
     coordarray = array[dim]
 
     # Handle dask arrays.
-    if isinstance(array, dask_array_type):
-        res = dask_array.map_blocks(coordarray, indx, dtype=indx.dtype)
+    if isinstance(array.data, dask_array_type):
+        res = array.map_blocks(
+            lambda a, b: a[b], coordarray, indx, dtype=indx.dtype
+        ).compute()
```

Comment:
What breaks if you don't `compute`?

Reply:
further down, this will break:

```python
# The dim is gone but we need to remove the corresponding coordinate.
del res.coords[dim]

# Copy attributes from argmin/argmax, if any
res.attrs = indx.attrs
```

Reply:
My dask-knowledge lacks, so I was not able to come up with a better solution, unfortunately

Reply:
ah this was quite broken. I just pushed a commit. Please see if that works. Clearly we need to add some tests with dask-backed objects.

Reply:
@dcherian Thanks, I'll have a look first thing next morning (CET).

Reply:
+1 for adding dask tests